libs/ui-toolkit/src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,79 @@
+import type { ReactNode } from 'react';
+import classnames from 'classnames';
+import { Button } from '../button';
+import { Icon } from '../icon';
+
+export type PaginationProps = {
+  className?: string;
+  hasPrevPage: boolean;
+  hasNextPage: boolean;
+  isLoading?: boolean;
+  children?: ReactNode;
+  onBack: () => void;
+  onNext: () => void;
+  onFirst?: () => void;
+  onLast?: () => void;
+};
+
+export const Pagination = ({
+  className,
+  hasPrevPage,
+  hasNextPage,
+  isLoading,
+  children,
+  onBack,
+  onNext,
+  onFirst,
+  onLast,
+}: PaginationProps) => {
+  return (
+    <div
+      className={classnames(
+        'flex gap-2 items-center justify-center',
+        className
+      )}
+    >
+      {onFirst && (
+        <Button
+          size="sm"
+          data-testid="goto-first-page"
+          disabled={isLoading || !hasPrevPage}
+          className="rounded-full w-[34px] h-[34px]"
+          onClick={onFirst}
+        >
+          <Icon name="double-chevron-left" ariaLabel="Back" />
+        </Button>
+      )}
+      <Button
+        size="sm"
+        data-testid="goto-previous-page"
+        disabled={isLoading || !hasPrevPage}
+        className="rounded-full w-[34px] h-[34px]"
+        onClick={onBack}
+      >
+        <Icon name="chevron-left" ariaLabel="Back" />
+      </Button>
+      {children}
+      <Button
+        size="sm"
+        data-testid="goto-next-page"
+        disabled={isLoading || !hasNextPage}
+        className="rounded-full w-[34px] h-[34px]"
```

Comment:
Nitpick (non-blocking): perhaps this value of 34px can be captured somewhere and reused for lines 51, 61 and 71

Reply:
done, made a variable for it